packages/core/src/base/cacheManagerKeyv.ts

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+"use strict";
+
+import fetch = require("minipass-fetch");
+import Keyv = require("keyv");
+import ssri = require("ssri");
+import url = require("url");
+
+import { ICacheManager } from "./cacheManager";
+
+/**
+ * Calculate a reasonable time to live for the response based on the request and response headers.
+ *
+ * @param response The response to calculate the TTL for
+ */
+const getTTL = (response: fetch.Response): number => {
+  const responseControl = response.headers
+    .get("cache-control")
+    ?.toLowerCase()
+    .trim()
+    .split(/\s*,\s*/);
+
+  if (responseControl) {
```

Comment:
Can we negate this condition and change the flow accordingly? to avoid nested ifs

Reply:
I wanted to, but I'm having a hard time that. We can't return if there isn't a cache-control header because we need to check for the expires header. But we still need to check the expires header AFTER if there's a cache-control header that doesn't contain anything interesting to us.